src/Compilers/CSharp/Test/Emit/Emit/DynamicAnalysis/DynamicInstrumentationTests.cs

```diff
@@ -859,6 +860,94 @@ File 1
         }
 
         [Fact]
+        public void LocalFunctionWithLambdaCoverage()
+        {
+            string source = @"
+using System;
+
+public class Program
+{
+    public static void Main(string[] args)                                  // Method 1
+    {
+        TestMain();
+        Microsoft.CodeAnalysis.Runtime.Instrumentation.FlushPayload();
+    }
+    
+    static void TestMain()                                                  // Method 2
+    {
+        new D().M1();
+    } 
+}
+
+public class D
+{
+    public void M1()                                                        // Method 4
+    {
+        L1();
+        void L1()
+        {
+            var f = new Func<int>(
+                () => 1
+            );
+
+            var f1 = new Func<int>(
+                () => 1
+            );
+
+            f();
+        }
+    }
+
+    // Method 5 is the synthesized instance constructor for D.
+}
+";
+
+            string expectedOutput = @"Flushing
+Method 1
+File 1
+True
+True
+True
+Method 2
+File 1
+True
+True
+Method 4
+File 1
+True
+True
+True
+True
+False
```

Comment:
Temporarily setting aside the larger question, this False is for the second lambda, which isn't called.

The mysterious "extra" True that appears for every called method represents entry to the method.

The corresponding resources test is easier to follow in terms of understanding which bits of text map to which spans, because fragments of text appear in the verification step.

What @jcouv is asking for is entirely reasonable, except that I don't know how to do it.